test_runner/fixtures/utils.py

```diff
@@ -86,7 +87,15 @@ def __init__(self, in_file, out_file, echo: bool, capture: bool):
             self.captured = ""
 
         def run(self):
+            first = capture_command
             for line in self.in_file:
+                if first:
+                    # do this only after receiving any input so that we can keep deleting empty files
+                    first = False
+                    # prefix the files with the command line so that we can later understand which file is for what command
+                    cmdline = " ".join(cmd)
+                    self.out_file.write(("# " + cmdline + "\n---\n").encode("utf-8"))
```

Comment:
I see you added the `--\n`. Aesthetically I'd like to keep the header clean, would you settle for removing the dashes, but keeping an empty line after header (in addition to the NL I added originally)?

Reply:
Sure, let's just keep an empty line between the command and the rest of output